jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlParser.java

```diff
@@ -162,13 +156,37 @@ public boolean isValidating()
         return _parser.isValidating();
     }
 
+    /**
+     * Load the specified URI as a catalog for entity mapping purposes.
+     *
+     * @param url the url to the catalog
+     */
+    public void addCatalog(URL url)
+    {
+        addCatalog(URI.create(url.toExternalForm()));
+    }
+
```

Comment:
Do we really need the URL form?

Reply:
The conversion from URL to URI needs to be somewhere, decided to keep it here.

Otherwise every `XmlParser.class.getResource(String)` style or `Loader.getResource(String)` style call (that returns URL) needs to do the conversion at that point.

Reply:
The use of `URL.create(url.toExternalForm())` makes this a runtime exception if there is an issue.
vs the use of `URL.toURI()` which is a checked exception.